Mathlib/Algebra/BigOperators/Group/Finset.lean

```diff
@@ -1219,36 +1222,48 @@ theorem prod_ite {s : Finset α} {p : α → Prop} {hp : DecidablePred p} (f g :
 #align finset.sum_ite Finset.sum_ite
 
 @[to_additive]
-theorem prod_ite_of_false {p : α → Prop} {hp : DecidablePred p} (f g : α → β) (h : ∀ x ∈ s, ¬p x) :
-    ∏ x ∈ s, (if p x then f x else g x) = ∏ x ∈ s, g x := by
-  rw [prod_ite, filter_false_of_mem, filter_true_of_mem]
-  · simp only [prod_empty, one_mul]
-  all_goals intros; apply h; assumption
+lemma prod_dite_of_false {p : α → Prop} {_ : DecidablePred p} (h : ∀ i ∈ s, ¬ p i)
```

Comment:
From the user point of view, it seems to me that the ordering `f,g,h` is more natural than `h,f,g`.

Reply:
Given that `h` is an hypothesis, it should come first. Indeed, all uses in mathlib fill in `h` but don't specify `f` and `g`.

Reply:
Good point.